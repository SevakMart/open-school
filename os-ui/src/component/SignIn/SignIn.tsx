import { useState, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import styles from './SignIn.module.scss';
import CloseIcon from '../../icons/Close';
import LinkedinIcon1 from '../../icons/Linkedin1';
import EmailIcon1 from '../../icons/Email1';
import { SIGN_UP } from '../../constants/Strings';
import Form from '../Forms/SignUpSignInForm';

const SignIn = ({ handleSignInClicks }:{handleSignInClicks(arg:string):void}) => {
  const {
    mainContainer, formContainer, headerContent, iconContent, alreadyHaveAccount, inputContent,
  } = styles;
  const navigate = useNavigate();
  const [isSignedIn, setIsSignedIn] = useState(false);
  const [succesfulSignInMessage, setSuccessfulSignInMessage] = useState('');

  const handleSignIn = (message:string) => {
    setIsSignedIn(true);
    setSuccessfulSignInMessage(message);
  };
  useEffect(() => {
    let timer:any;
    if (isSignedIn) {
      timer = setTimeout(() => navigate('/choose_categories'), 3000);
    }
    return () => clearTimeout(timer);
  }, [isSignedIn]);

  return (
    <div className={mainContainer}>
      <div className={formContainer}>
        { isSignedIn ? <h3>{succesfulSignInMessage}</h3>
          : (
            <>
              <CloseIcon handleClosing={() => handleSignInClicks('closeButton')} />
              <div className={headerContent}>
                <h2>Sign In!</h2>
                <div className={iconContent}>
                  <button type="button"><LinkedinIcon1 /></button>
                  <button type="button"><EmailIcon1 /></button>
                </div>
                <p>Or</p>
              </div>
              <form className={inputContent}>
                <Form
                  formType="signIn"
                  handleSignIn={handleSignIn}
                />
              </form>
              <p className={alreadyHaveAccount}>
                {'Don\'t Have An Account?'}
                <span><button type="button" onClick={() => handleSignInClicks('signUp')}>{SIGN_UP}</button></span>
              </p>
            </>
          )}
      </div>
    </div>
  );
};
export default SignIn;
