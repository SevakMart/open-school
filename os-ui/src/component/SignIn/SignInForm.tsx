import SignInDefault from './Default/Default';

const SignInForm = ({ signInForm, handleSignIn, forgotPasswordHandler }:
    {signInForm:string, handleSignIn:(message:string)=>void, forgotPasswordHandler:()=>void}) => {
  const handleForgotPassword = () => forgotPasswordHandler();

  return (
    <>
      {
        signInForm === 'default'
          ? (
            <SignInDefault
              handleSignIn={(message:string) => handleSignIn(message)}
              forgotPasswordFunc={handleForgotPassword}
            />
          ) : null
        }
    </>
  );
};
export default SignInForm;
