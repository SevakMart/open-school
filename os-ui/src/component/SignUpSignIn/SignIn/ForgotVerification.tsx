import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { useDispatch } from 'react-redux';
import { validateEmail } from '../../../helpers/EmailValidate';
import { addEmail } from '../../../redux/Slices/ForgotPasswordEmailSlice';
import { openModalWithSuccessMessage } from '../../../redux/Slices/PortalOpenStatus';
import authService from '../../../services/authService';
import { Types } from '../../../types/types';
import Button from '../../Button/Button';
import { Input } from '../../Input/Input';
import Header from '../Header/Header';

type ForgotVerificationProps = {
	forgotVerficationEmail?: string
}

const ForgotVerification = (props: ForgotVerificationProps) => {
  const { forgotVerficationEmail } = props;

  const { t } = useTranslation();
  const dispatch = useDispatch();
  const [email, setEmail] = useState(forgotVerficationEmail || '');
  const [emailError, setEmailError] = useState('');

  const handleInputChange = (e:React.SyntheticEvent) => {
    setEmail((e.target as HTMLInputElement).value);
  };

  const handleFormOnClick = (e: React.SyntheticEvent) => {
    e.preventDefault();
  };

  const handleSubmit = (event: React.KeyboardEvent<HTMLInputElement>): void => {
    if (event.key === 'Enter') {
	  event.preventDefault();
	  handleFormOnClick(event);
    }
  };

  const sendForgotVerification = () => {
    const emailError = validateEmail(email);
    if (!emailError) {
      authService.sendForgotVerificationRequest({ email })
        .then((response) => {
          if (response.status === 200) {
            const successMessage = t('Verification Email Sent!');
            dispatch(openModalWithSuccessMessage({
              buttonType: Types.Button.SUCCESS_MESSAGE,
              withSuccessMessage: successMessage,
              isForgotVerificationMessage: true,
            }));
            dispatch(addEmail(email));
            setEmailError('');
          }
        });
    } else setEmailError(emailError);
  };

  return (
    <>
      <Header
        mainTitle={t('Your verification link has expired, enter mail to verify your account')}
        shouldRemoveIconContent
        isForgotPasswordContent={false}
        isVerificationContent={false}
      />
      <div style={{ margin: '0 80px' }}>
        <Input.EmailInput
          textName="email"
          labelText={t('form.labels.email')}
          errorMessage={emailError}
          placeholderText={t('form.placeholder.email')}
          value={email}
          handleInputChange={handleInputChange}
          handleEnterPress={handleSubmit}
        />
        <Button.FormButton className={['formButton']} onClick={sendForgotVerification}>
          {t('Resend')}
        </Button.FormButton>
      </div>
    </>
  );
};

export default ForgotVerification;
