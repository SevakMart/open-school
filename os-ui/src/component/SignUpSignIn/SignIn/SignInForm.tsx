import SignInDefault from './Default/Default';

const SignInForm = ({ signInForm, handleSignIn }:{signInForm:string, handleSignIn:()=>void}) => (
  <>
    {signInForm === 'default' && <SignInDefault handleSignIn={() => handleSignIn()} />}
  </>
);
export default SignInForm;
