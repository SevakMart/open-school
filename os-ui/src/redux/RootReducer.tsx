import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
