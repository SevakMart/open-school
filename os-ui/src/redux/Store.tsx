import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
