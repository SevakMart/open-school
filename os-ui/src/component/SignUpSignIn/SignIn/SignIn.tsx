import { useState, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import SignInForm from './SignInForm';
import Header from '../Header/Header';
import Footer from '../Footer/Footer';
import { Types } from '../../../types/types';
import styles from './SignIn.module.scss';

const SignIn = ({ handleSignInClicks }:{handleSignInClicks(arg:string):void}) => {
  const { t } = useTranslation();
  const {
    mainContainer, formContainer, headerContent, iconContent, alreadyHaveAccount,
  } = styles;
  const navigate = useNavigate();
  const [isSignedIn, setIsSignedIn] = useState(false);
  const [forgotPasswordIsSet, setForgotPasswordIsSet] = useState(false);

  const handleSignIn = () => {
    setIsSignedIn(true);
  };

  const forgotPassword = () => {
    setForgotPasswordIsSet(true);
  };

  const returnToSignInForm = () => {
    setForgotPasswordIsSet(false);
    setIsSignedIn(false);
  };

  useEffect(() => {
    if (isSignedIn) {
      navigate('/categories/subcategories');
    }
  }, [isSignedIn]);

  return (
    <>
      <Header
        mainTitle={t('string.signIn.title')}
        shouldRemoveIconContent={false}
        isForgotPasswordContent={false}
      />
      <SignInForm
        signInForm="default"
        handleSignIn={handleSignIn}
        forgotPasswordHandler={forgotPassword}
      />
      <Footer
        mainText={t('string.signIn.dontHaveAccount')}
        buttonType={Types.Button.SIGN_UP}
        buttonText={t('button.homePage.signUp')}
      />
    </>
  );
};
export default SignIn;
