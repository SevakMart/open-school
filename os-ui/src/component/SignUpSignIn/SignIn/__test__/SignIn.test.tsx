import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { store } from '../../../../redux/Store';
import SignIn from '../SignIn';

/* const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

describe('Create test cases for SignIn component', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <SignIn
          handleSignInClicks={() => null}
        />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
*/
