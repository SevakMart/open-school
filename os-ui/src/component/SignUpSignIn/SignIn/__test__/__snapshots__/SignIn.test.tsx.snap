// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test cases for SignIn component Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="closeIcon"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7,7 L17,17 M7,17 L17,7"
        fill="none"
        stroke="#000"
        stroke-width="2"
      />
    </svg>
  </div>
  <div
    class="headerContent"
  >
    <h2>
      string.signIn.title
    </h2>
    <div
      class="iconContent"
    >
      <div
        class="icon"
      >
        <a
          href=""
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            class="icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M847.7 112H176.3c-35.5 0-64.3 28.8-64.3 64.3v671.4c0 35.5 28.8 64.3 64.3 64.3h671.4c35.5 0 64.3-28.8 64.3-64.3V176.3c0-35.5-28.8-64.3-64.3-64.3zm0 736c-447.8-.1-671.7-.2-671.7-.3.1-447.8.2-671.7.3-671.7 447.8.1 671.7.2 671.7.3-.1 447.8-.2 671.7-.3 671.7zM230.6 411.9h118.7v381.8H230.6zm59.4-52.2c37.9 0 68.8-30.8 68.8-68.8a68.8 68.8 0 1 0-137.6 0c-.1 38 30.7 68.8 68.8 68.8zm252.3 245.1c0-49.8 9.5-98 71.2-98 60.8 0 61.7 56.9 61.7 101.2v185.7h118.6V584.3c0-102.8-22.2-181.9-142.3-181.9-57.7 0-96.4 31.7-112.3 61.7h-1.6v-52.2H423.7v381.8h118.6V604.8z"
            />
          </svg>
        </a>
      </div>
      <div
        class="icon"
      >
        <svg
          class="icon"
          fill="currentColor"
          height="1em"
          role="img"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title />
          <path
            d="M24 5.457v13.909c0 .904-.732 1.636-1.636 1.636h-3.819V11.73L12 16.64l-6.545-4.91v9.273H1.636A1.636 1.636 0 0 1 0 19.366V5.457c0-2.023 2.309-3.178 3.927-1.964L5.455 4.64 12 9.548l6.545-4.91 1.528-1.145C21.69 2.28 24 3.434 24 5.457z"
          />
        </svg>
      </div>
    </div>
    <p>
      string.signIn.or
    </p>
  </div>
  <form
    class="inputContent"
  >
    <div
      class="EmailInputContainer"
    >
      <label
        for="email"
      >
        form.labels.email
        <span
          style="color: red;"
        >
          *
        </span>
      </label>
      <input
        data-testid="emailInput"
        id="email"
        name="email"
        placeholder="form.placeholder.email"
        required=""
        tabindex="-1"
        type="email"
        value=""
      />
    </div>
    <div
      class="PasswordInputContainer"
    >
      <label
        for="psd"
      >
        form.labels.psd.default
        <span
          style="color: red;"
        >
          *
        </span>
      </label>
      <div
        class="PasswordInputFieldWithIcon"
      >
        <input
          data-testid="psdInput"
          id="psd"
          name="psd"
          placeholder="form.placeholder.psd.default"
          required=""
          tabindex="0"
          type="password"
          value=""
        />
        <i
          style="position: absolute; right: 15px; top: 0px; bottom: 0px; margin: auto; height: 1rem;"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5zm-63.57-320.64L836 122.88a8 8 0 0 0-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 0 0 0 11.31L155.17 889a8 8 0 0 0 11.31 0l712.15-712.12a8 8 0 0 0 0-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 0 0-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 0 1 146.2-106.69L401.31 546.2A112 112 0 0 1 396 512z"
            />
            <path
              d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 0 0 227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 0 1-112 112z"
            />
          </svg>
        </i>
      </div>
      <h4
        class="inputErrorField"
      />
    </div>
    <p
      class="forgotPassword"
    >
      string.signIn.forgotPsd
    </p>
    <button
      class="formButton"
      type="submit"
    >
      button.homePage.signIn
    </button>
  </form>
  <p
    class="alreadyHaveAccount"
  >
    string.signIn.dontHaveAccount
    <span>
      <button
        class="account"
        type="button"
      >
        button.homePage.signUp
      </button>
    </span>
  </p>
</DocumentFragment>
`;
